packages/wallet/src/errors.ts

```diff
@@ -0,0 +1,55 @@
+import { inspect } from "node:util";
+
+export class InvalidPayloadError extends Error {
+    constructor(public payload: unknown) {
+        super(`Invalid payload: ${inspect(payload)}`);
+    }
+}
+
+export class RelayError extends Error {
+    constructor() {
+        super(
+            `You need to call the method relayDAppAddress from DAppAddressRelay__factory.`,
+        );
+    }
+}
+
+export class InsufficientBalanceError extends Error {
+    constructor(user: string, token: string, amount: bigint, tokenId?: bigint) {
+        super(
+            `insufficient balance of user ${user} of token ${token}: ${amount.toString()}${
+                tokenId ? ` of tokenId ${tokenId}` : ""
+            }`,
+        );
+    }
+}
+
+export class WalletUndefinedError extends Error {
+    constructor(address: string) {
+        super(`wallet of user ${address} is undefined`);
+    }
+}
+
+export class TokenFromUserNotFound extends Error {
+    constructor(user: string, token: string, tokenId: bigint) {
+        super(`user ${user} does not have token ${tokenId} of token ${token}`);
+    }
+}
+
+export class ArrayNoSameLength extends Error {
+    constructor(first: string, second: string) {
+        super(`${first} and ${second} must have the same length`);
```

Comment:
are we raising this in what situation?

Reply:
see: https://github.com/tuler/deroll/pull/62#discussion_r1546686258